backend/tests/integration/files/test_api_s3.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+import pytest
+from botocore.exceptions import ClientError
+from mock import patch
+
+from inspirehep.files.api import S3
+
+
+def test_upload_file(base_app, appctx):
+    data = "this is my file data"
+    key = "e50c2ea2d26571e0c5a3411e320586289fd715c2"
+    filename = "file.txt"
+    mimetype = "text/*"
+    acl = "public-read"
+    expected_extraargs = {
+        "ContentType": mimetype,
+        "ACL": acl,
+        "ContentDisposition": f'attachment; filename="{filename}"',
+    }
+    s3 = S3()
+    with patch.object(s3, "client") as mock_client:
```

Comment:
have you looked into using https://github.com/spulec/moto instead of mocking the client manually?